### PR TITLE
More newPlot calls in tests

### PR DESCRIPTION
**Pull request by archmoj**
Follow up of #5370.

@plotly/plotly_js 

cc: #5395

### CODE REVIEW

File: test/jasmine/tests/geo_test.js [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ describe('Test geo interactions', function() {
                     }
 
                     gd.calcdata = undefined;
-                    Plotly.plot(gd);
+                    Plotly.newPlot(gd, gd.data, gd.layout);
```
Comment (alexcjohnson) [inline test/jasmine/tests/geo_test.js:1219]: At some point we recommended this pattern - `gd.calcdata = undefined` before a `Plotly.plot` call - if you wanted to update the figure object in place and let the graph know it should redo its `calc` step.

If you replace `plot` with `newPlot` in these cases, you're doing too much cleaning so we're not testing the intended behavior.

Instead, these cases should turn into `Plotly.react` with a `layout.datarevision` value that changes every time (set its value to `i` perhaps). And you can remove the `gd.calcdata = undefined`.

Comment (archmoj) [reply test/jasmine/tests/geo_test.js]: Thanks for the info. 
Addressed in https://github.com/plotly/plotly.js/pull/5393/commits/981cc40fe02469db4bfa2ffe3de8122401733aff.

File: test/jasmine/tests/contour_test.js [hunk 2]
```diff
@@ -639,13 +639,13 @@ describe('contour hover', function() {
...
 
         it('should not display hover on missing data and hoverongaps is disabled', function() {
-            var pt = _hover(gd, 10, 100)[0];
-
             var hoverData;
             gd.on('plotly_hover', function(data) {
                 hoverData = data;
             });
 
+            var pt = _hover(gd, 10, 100)[0];
```
Comment (alexcjohnson) [inline test/jasmine/tests/contour_test.js:647]: nice catch 🤦

**Review by alexcjohnson — APPROVED** [review]
💃 Beautiful!

**Pull request merged**